client/src/extension.ts

```diff
@@ -19,9 +19,9 @@ import { telemetryService } from './telemetry';
 
 const outputChannel : vscode.OutputChannel = vscode.window.createOutputChannel(`SLDS`);
 
-export function activate(context: vscode.ExtensionContext) {
+export async function activate(context: vscode.ExtensionContext) {
 	// Telemetry service
-	telemetryService.initializeService(context);
+	await telemetryService.initializeService(context);

```

Comment:
you are doing this because you want the `telemetryService` to initialize first?

Reply:
yupp. there is quite a bit going on now in the `initializeService` function now since we checking if the the user disabled telemetry via sfdx cli so this is to ensure that all happens before moving forward.